[DOCS]Add Catalog Promotion documentation (#13084)
Sylius/Sylius: docs/book/products/catalog_promotions.rst

.. index::
   single: Catalog Promotions

Catalog Promotions
==================

The **Catalog Promotions** system in **Sylius** is a new way of dealing with promotions on multiple products.
If you get used to `Cart Promotions </book/orders/cart-promotions>` this will be something familiar to you.

> It is managed by combination of promotion rules and actions, where you can specify on which f.e. products or taxons

Reviewer: ```suggestion
It is managed by combination of promotion rules and actions, where you can specify on which e.g. products or taxons
```

> you can specify the Catalog Promotion with your custom actions as well as actions like percentage or value discount.

Reviewer 2: ```suggestion
you can specify the Catalog Promotion with your custom actions as well as actions like percentage discount.
```
it is not true yet.

You can assign the needed channels too.

Catalog Promotion Parameters
----------------------------

> Catalog Promotion has a few basic parameters that represents it - a unique ``code`` and ``name``:

Reviewer: ```suggestion
Catalog Promotion has a few basic parameters that represent it - a unique ``code`` and ``name``:
```

.. note::

    The parameter ``code`` should contain only letters, numbers, dashes and underscores.
    We encourage you to use ``snake_case`` codes.

.. code-block:: bash

    {
        "code": "catalog_promotion" # unique
        "name": "catalog promotion"
        # ...
    }

Rest of the fields are used for configuration:

* **Channels** are used to define channels on which given promotion is applied:

.. code-block:: bash

    {
        #...
        "channels": [
            "/api/v2/admin/channels/FASHION_WEB", #IRI
            "/api/v2/admin/channels/HOME_WEB"
        ]
        # ...
    }

* **Rules** are used to define rules on which the catalog promotion will work:

.. code-block:: bash

    {
        #...
        "rules": [
            {
                "type": "for_variants",
                "configuration": {
                    "variants": [

>                         "Everyday_white_basic_T_Shirt-variant-1", #Variant Code

Author: Do you have something in mind?

    {
        #...
        "translations": {
            "en_US": {
                "label": "Summer discount",
                "description": "The grass so green, the sun so bright. Life seems a dream, no worries in sight.",
                "locale": "en_US" #Locale Code
                }
            }
        }
        # ...
    }

How to create a Catalog Promotion?
----------------------------------

After we get to know with some basics of Catalog Promotion let's see how we can create one:

* **API** The common use case is to make it through API, first you need to authorize yourself as an admin (you don't want to let a guest create it - don't you?).

.. tip::

    Check this doc `Authorization </book/api/authorization>` if you are having trouble with login in.

And let's call the POST endpoint to create very basic catalog promotion:

.. code-block:: bash

    curl -X 'POST' \
      'https://hostname/api/v2/admin/catalog-promotions' \
      -H 'accept: application/ld+json' \
      -H 'Authorization: Bearer authorizationToken' \
      -H 'Content-Type: application/ld+json' \
      -d '{
        "code": "t_shirt_promotion",
        "name": "T-shirt Promotion"
        }'

> 

Reviewer: ```suggestion
```

We can also make it programmatically:

.. code-block:: php

    /** @var CatalogPromotionInterface $catalogPromotion */
    $catalogPromotion = $this->container->get('sylius.factory.catalog_promotion')->createNew();
    $catalogPromotion->setCode('t_shirt_promotion');
    $catalogPromotion->setName('T-shirt Promotion');

    $catalogPromotion->setCurrentLocale('en_US');
    $catalogPromotion->setFallbackLocale('en_US');
    $catalogPromotion->setLabel('T-shirt Promotion');
    $catalogPromotion->setDescription('T-shirt Promotion description');

    $catalogPromotion->addChannel('FASHION_WEB');

    /** @var CatalogPromotionRuleInterface $catalogPromotionRule */
    $catalogPromotionRule = $this->catalogPromotionRuleExampleFactory->create($rule);
    $catalogPromotionRule->setCatalogPromotion($catalogPromotion);
    $catalogPromotion->addRule($catalogPromotionRule);

    /** @var CatalogPromotionActionInterface $catalogPromotionAction */
    $catalogPromotionAction = $this->catalogPromotionActionExampleFactory->create($action);
    $catalogPromotionAction->setCatalogPromotion($catalogPromotion);
    $catalogPromotion->addAction($catalogPromotionAction);

> And now you should be able to see created Catalog Promotion. You can check if it exist like in last example (with GET endpoint).

Reviewer: ```suggestion
And now you should be able to see created Catalog Promotion. You can check if it exists like in the last example (with GET endpoint).
```

If you look into ``product-variant`` endpoint in shop you should see now that chosen variants have lowered price and added field ``appliedPromotions``:

.. code-block:: bash

    curl -X 'GET' \
    'https://hostname/api/v2/shop/product-variant/Everyday_white_basic_T_Shirt-variant-1'

.. code-block:: bash

    # response content

> 

Reviewer: ```suggestion
```
